### PR TITLE
Make hotkey toggle CommandPal on/off

### PR DESCRIPTION
**Pull request by rouilj**
Make the hotkey able to toggle command pal on and dismiss it as well.

Inspired by

https://plutoapp.xyz/blog/post/designing-a-command-palette/#togglable

from: https://commandpalette.org

Basic reason is if a user accidentally hits it, they can hit it again to get out of it. Otherwise they have to figure out how to dismiss it.

Three changes:

1) add data-id=cp-SearchField to search input. When I first coded this
   it only worked with hotkeysGlobal: true. It would not work with
   hotkeysglobal: false.

   It finally dawned on me that the focus was in the search input when
   I sent the dismissal hotkey. hotkeys.js ignores hotkeys in an input
   by default. To fix this I had to find a way to uniquely identify
   this input to I could make hotkeys active. The normal id is a
   unique random string, so no joy there.

2) changed onMount to look at current state of showModal and call
   onClosed if true.

3) rewrote the hotkeys.js filter. Now I understand it. Wish I didn't
   8-).  When the filter function is defined, I make it a closure over
   BindToInputsToo. The new filter does 1 of three things:

      * if BindToInputsToo is true, it return true, so hotkeys will process it
      * if BindToInputsToo is false, check to see what the activeElement is. If its data-id is cp-SearchField return true so keys is processed. * otherwise look to see if it's a ContentEditable area, or a writable input or select. If so return false, else return true

Also I used https://github.com/jaywcjlove/hotkeys/issues/321 to figure out how to create a filter, it's not correct.

  tagname = target.tagName

not

  tagname = target

I don't know if that's why hotkeysGlobal didn't work originally.

Also the default scope is all. So no need to set it.

All that matters is that the filter returns true to process the key or false to ignore it. Setting the scope in the filter probably changes the lookup table used to find matching keys.

### CODE REVIEW

File: src/shortcuts.js [hunk 1]
```diff
@@ -3,15 +3,30 @@ import hotkeys from "hotkeys-js";
...
 export function initShortCuts(bindToInputsToo) {
-  if (bindToInputsToo) {
-    /* 
-    Allows binding to input, select and textarea
-    https://stackoverflow.com/questions/59855852/input-blocks-hotkeys
-    Appears to not work. Setting scope to "all" does work. // rouilj
-    */
-    hotkeys.filter = function(event){
-      hotkeys.setScope('all');
-      return true;
```
Comment (benwinding) [inline src/shortcuts.js:14]: A shame this had to go, this was much simpler 😭

File: src/App.svelte [hunk 2]
```diff
@@ -38,9 +38,14 @@
   onMount(() => {
     initShortCuts(hotkeysGlobal);
     setMainShortCut(hotkey, async () => {
-      showModal = true;
-      selectedIndex = 0;
-      dispatch("opened");
+      if (showModal) {
+	onClosed()
+      } else {
+	focusedElement = document.activeElement
```
Comment (benwinding) [inline src/App.svelte:44]: This seems to break the build, as the `focusedElement` variable isn't declared anywhere, what's it for?
(_merge current master into this branch_)

**Review by benwinding — APPROVED** [review]
> Basic reason is if a user accidentally hits it, they can hit it again to get out of it. Otherwise they have to figure out how to dismiss it.

I like it, it's much more user-friendly to use the same shortcut to toggle or the `escape` key.

Learning so much about _Command Palettes_ through these rabbit holes 👍

**Comment by rouilj** [comment]
> I like it, it's much more user-friendly to use the same shortcut to toggle or the `escape` key.

Not my idea:

  https://destiner.io/blog/post/designing-a-command-palette/

**Pull request merged**